crest/Assets/Crest/Crest/Shaders/OceanInputs/AnimWavesScaleByFactor.shader

```diff
@@ -0,0 +1,123 @@
+﻿// Crest Ocean System
+
+// This file is subject to the MIT License as seen in the root of this folder structure (LICENSE)
+
+// This writes straight into the displacement texture and sets the water height to the y value of the geometry.
+
+Shader "Crest/Inputs/Animated Waves/Scale By Factor"
+{
+	Properties
+	{
+		[Enum(ColorWriteMask)] _ColorWriteMask("Color Write Mask", Int) = 15
+
+		// Scale the waves. Zero is no waves and one leaves waves untouched.
+		_Scale("Scale", Range(0, 1)) = 0.35
+
+		// Use the texture instead of the scale value.
+		[Toggle] _Texture("Apply Texture", Float) = 0
+		_MainTex("Texture", 2D) = "black" {}
+
+		// Inverts the scale value.
+		[Toggle] _Invert("Invert", Float) = 0
```

Comment:
Happy for this to go in but would be curious what the use case is? Just checking it's needed.

Reply:
Makes it more flexible with available textures. I was using the standard particle texture for testing which is 1 (no scaling) in the middle and thought it might be useful to have. Happy to remove it if you think it isn't worth keeping.

Reply:
No that's a fair point, good like that then